source/renderer/app/containers/wallet/dialogs/send-confirmation/Container.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useMemo, useCallback, useState } from 'react';
+import { inject, observer } from 'mobx-react';
+import compose from 'lodash/fp/compose';
+import { getNonZeroAssetTokens } from '../../../../utils/assets';
+import { ContainerProps as Props, SubmitPayload } from './types';
+import { WalletSendConfirmationDialogView } from './View';
+
+export const Containter = ({
+  actions,
+  stores,
+  amount,
+  selectedAssets,
+  assetsAmounts,
+  receiver,
+  totalAmount,
+  onExternalLinkClick,
+  transactionFee,
+  hwDeviceStatus,
+  isHardwareWallet,
+  formattedTotalAmount,
+}: Props) => {
+  // @ts-ignore ts-migrate(2339) FIXME: Property 'isFlight' does not exist on type 'typeof... Remove this comment to see the full error message
+  const { isFlight } = global;
+  const {
+    assets: { getAsset },
+    wallets: { sendMoneyRequest, active: activeWallet },
+    hardwareWallets: {
+      _resetTransaction: resetHardwareWalletTransaction,
+      sendMoneyRequest: sendMoneyExternalRequest,
+      isTransactionPending,
+      checkIsTrezorByWalletId,
+      initiateTransaction,
+    },
+  } = stores;
+  const {
+    assets: { onCopyAssetParam },
+    dialogs: { closeActiveDialog },
+    hardwareWallets: { sendMoney: hardwareWalletsSendMoney },
+    wallets: { sendMoney: walletsSendMoney },
+  } = actions;
+
+  if (!activeWallet) {
+    throw new Error('Active wallet required for WalletSendPage.');
+  }
+
+  const [areTermsAccepted, setAreTermsAccepted] = useState(false);
+  const assetTokens = useMemo(
+    () => getNonZeroAssetTokens(activeWallet.assets.total, getAsset),
+    [activeWallet.assets.total, getAsset]
+  );
+  const isSubmitting =
+    (!isHardwareWallet && sendMoneyRequest.isExecuting) ||
+    (isHardwareWallet &&
+      (sendMoneyExternalRequest.isExecuting || isTransactionPending));
+
+  const error = isHardwareWallet
+    ? sendMoneyExternalRequest.error
+    : sendMoneyRequest.error;
+
+  const onSubmitCb = useCallback(
+    (values: SubmitPayload) => {
+      if (values.isHardwareWallet) {
+        hardwareWalletsSendMoney.trigger({});
```

Comment:
Would be possible to make the trigger argument optional so you don't have to pass `{}`?

Reply:
Updated, thanks!